src/response.js

```diff
@@ -23,12 +39,18 @@ export default class Response extends Body {
 	constructor(body = null, options = {}) {
 		super(body, options);
 
-		// eslint-disable-next-line no-eq-null, eqeqeq, no-negated-condition
-		const status = options.status != null ? options.status : 200;
+		// if status is not provided or is null, let's use default 200 value
+		// otherwise, lets check if the value is in correct range
+		const status =	(typeof options.status === 'undefined' || options.status === null) ? 200 :
+			validateStatusValue(options.status);
 
 		const headers = new Headers(options.headers);
 
-		if (body !== null && !headers.has('Content-Type')) {
+		if (status === 204 && typeof body !== 'undefined' && body !== null) {
```

Comment:
I think we also need `205` and `304`, as per [the standard](https://fetch.spec.whatwg.org/#null-body-status:~:text=A%20null%20body%20status%20is%20a%20status%20that%20is%20101%2C%20103%2C%20204%2C%20205%2C%20or%20304.)